packages/api/src/utils/psa.js

```diff
@@ -56,19 +61,191 @@ export const psaStatusesToDBStatuses = (statuses) => {
 }
 
 // Error messages
-// TODO: Refactor errors
-export const ERROR_CODE = 400
-export const ERROR_STATUS = 'INVALID_PIN_DATA'
-export const INVALID_CID = 'Invalid cid'
-export const INVALID_MATCH = 'Match should be a string (i.e. "exact", "iexact", "partial", "ipartial")'
-export const INVALID_META = 'Meta should be an object with string values'
-export const INVALID_NAME = 'Name should be a string'
-export const INVALID_ORIGINS = 'Origins should be an array of strings'
-export const INVALID_REQUEST_ID = 'Request id should be a string containing digits only'
-export const INVALID_REPLACE = 'Existing and replacement CID are the same'
-export const INVALID_STATUS = 'Status should be a list of "queued", "pinning", "pinned", or "failed"'
-export const INVALID_TIMESTAMP = 'Should be a valid timestamp'
-export const INVALID_LIMIT = 'Limit should be a number'
-export const REQUIRED_CID = 'CID is required'
-export const REQUIRED_REQUEST_ID = 'Request id is required'
-export const UNPERMITTED_MATCH = 'Match should be "exact", "iexact", "partial", or "ipartial"'
+export const DATA_NOT_FOUND = 'Requested data was not found.'
+export const INVALID_CID = 'The CID provided is invalid.'
+export const INVALID_REPLACE = 'Existing and replacement CID are the same.'
+export const INVALID_REQUEST_ID = 'Request id should be a string.'
+export const PINNING_FAILED = 'PSA_PINNING_FAILED'
+export const REQUIRED_CID = 'CID is required.'
+export const REQUIRED_REQUEST_ID = 'Request id is required.'
+
+export const DEFAULT_PIN_LISTING_LIMIT = 10
+export const MAX_PIN_LISTING_LIMIT = 1000
+
+// Validation schemas
+const listPinsValidator = new Validator({
+  type: 'object',
+  required: [],
```

Comment:
is status required? I was looking into nft.storage schema and it is required there. In the spec https://ipfs.github.io/pinning-services-api-spec/#tag/pins/paths/~1pins/get it is mentioned as `Array of strings (Status) non-empty`

Reply:
@lidel Can you please let us know your thoughts here?

Reply:
iirc an explicit `status` is required to remove any ambiguity – the field is marked as `non-empty` in the spec for that reason.
if it is missing, you should return HTPP 400 BAD_REQUEST with `error.description` informing user that explicit `status` has to be provided.


The idea here is that the decision around implicit defaults should happen on a client, not the server.
For example, `ipfs pin remote ls --service=foo` without `--status`  defaults to `pinned` and sends `pins?status=pinned` on the wire.